spras/btb.py

```diff
@@ -0,0 +1,188 @@
+# need to define a new btb class and contain the following functions
+# - generate_inputs
+# - run
+# - parse_output
+
+import warnings
+from pathlib import Path
+
+import pandas as pd
+
+from spras.containers import prepare_volume, run_container
+# from spras.interactome import (
+#     convert_undirected_to_directed,
+#     reinsert_direction_col_directed,
+# )
+# what type of directionality does btb support?
+
+from spras.prm import PRM
+
+__all__ = ['BowtieBuilder']
+
+class BowtieBuilder(PRM):
+    required_inputs = ['sources', 'targets', 'edges']
+
+    #generate input taken from meo.py beacuse they have same input requirements
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in BowtieBuilder.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+        print("FILEMAP NAME: ", filename_map)
+        print("DATA HEAD: ")
+        print( data.node_table.head())
+        print("DATA INTERACTOME: ")
+        print(data.interactome.head())
+
+        # Get sources and write to file, repeat for targets
+        # Does not check whether a node is a source and a target
+        for node_type in ['sources', 'targets']:
+            nodes = data.request_node_columns([node_type])
+            if nodes is None:
+                raise ValueError(f'No {node_type} found in the node files')
+
+            # TODO test whether this selection is needed, what values could the column contain that we would want to
+            # include or exclude?
+            nodes = nodes.loc[nodes[node_type]]
```

Comment:
Let's check how BTB handles these cases - we might need to put in some checks here.

1. What happens if either the sources or the targets file is empty?
2. What happens if the same node appears in the sources or targets file?

Reply:
If BTB catches these issues and throws an error that's fine, I think. We just want to make sure that it does handle these issues.

Reply:
I'm a bit unclear on what you mean/what we should do about this.

Reply:
https://github.com/Reed-CompBio/BowTieBuilder-Algorithm/issues/1